src/test/java/com/pkb/unit/tracker/TrackerRestartTest.java

```diff
@@ -68,4 +71,42 @@ public void restartTrackerDoesntEmitWhenUnitFailsToRestart() {
         // THEN
         testRestartedObserver.assertEmpty();
     }
+
+    @Test
+    public void restartTrackerSubscribesToCorrectTarget() {
+        // GIVEN
+        setupComputationAndIOTestScheduler();
+
+        // WHEN
+        String unit1id = "unit1";
+        FakeUnit unit1 = new FakeUnit(unit1id, bus);
+        String unit2id = "unit2";
+        FakeUnit unit2 = new FakeUnit(unit2id, bus);
+        unit1.addDependency(unit2id);
+        unit1.enable();
+        unit2.completeStart();
+        unit1.completeStart();
+        testScheduler.triggerActions();
+
+        // THEN
+        assertLatestState(systemState().addUnits(
+                unit(unit1id).withState(STARTED).withDesiredState(ENABLED).withDependencies(unit2id),
+                unit(unit2id).withState(STARTED).withDesiredState(UNSET))
+                .build());
+
+        // WHEN
+        TestObserver<Boolean> testRestartedObserver = Tracker.unitRestarted(bus, unit1id).test();
+        unit2.stop();
+        unit2.completeStop();
+        unit1.completeStop();
+        unit2.start();
+        unit1.failStart();
+        testScheduler.triggerActions();
+
```

Comment:
You probably also want to assert the system state here too; unit2 should have successfully started (proving that a unit started, but the restartedObserver didn't trigger)

Reply:
ok, fixing